solidity/dashboard/src/services/tokens-page.service.js

```diff
@@ -374,11 +369,87 @@ const getGranteeDelegations = async (
     false
   )
 
-  return await getDelegations(
+  const [delegations, undelegations] = await getDelegations(
     allOperators,
     initializationPeriod,
     undelegationPeriod
   )
+
+  return [delegations, undelegations, granteeGrants]
+}
+
+export const getCopiedDelegations = async (
+  ownerOrGrantee,
+  grantIds,
+  initializationPeriod,
+  undelegationPeriod
+) => {
+  const {
+    stakingPortBackerContract,
+    grantContract,
+    stakingContract,
+  } = await ContractsLoaded
+
+  const operatorsToCheck = (
+    await stakingPortBackerContract.getPastEvents("StakeCopied", {
+      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingPortBackerContract,
+      filter: { owner: ownerOrGrantee },
+    })
+  ).map((_) => _.returnValues.operator)
+
+  // No delegations
+  if (isEmptyArray(operatorsToCheck)) {
+    return await getDelegations({}, initializationPeriod, undelegationPeriod)
+  }
+
+  // We only want operators for whom the delegation has not been paid back.
+  const operatorsOfPortBacker = await getOperatorsOfOwner(
+    stakingPortBackerContract.options.address,
+    operatorsToCheck
+  )
+
+  // Scan `OperatorStaked` event by operator(indexed param) to get authorizer and beneeficiary.
+  const operatorToDetails = (isEmptyArray(operatorsOfPortBacker)
+    ? []
+    : await stakingContract.getPastEvents("OperatorStaked", {
+        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
+        filter: { operator: operatorsOfPortBacker },
+      })
+  ).reduce((reducer, _) => {
+    reducer[_.returnValues.operator] = _.returnValues
+    reducer[_.returnValues.operator].isCopiedStake = true
+
+    return reducer
+  }, {})
+
+  // Get operators from delegations created from a grant.
+  let tokenGrantStakingEvents = []
+  if (!isEmptyArray(grantIds)) {
+    tokenGrantStakingEvents = await grantContract.getPastEvents(
+      "TokenGrantStaked",
+      {
+        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.grantContract,
+        filter: { grantId: grantIds },
+      }
+    )
+  }
+
+  for (const grantStakedEvent of tokenGrantStakingEvents) {
+    const operator = grantStakedEvent.returnValues.operator
+    const grantId = grantStakedEvent.returnValues.grantId
+    if (operatorToDetails.hasOwnProperty(operator)) {
+      operatorToDetails[operator].grantId = grantId
+      operatorToDetails[operator].isFromGrant = true
+    }
+  }
```

Comment:
Maybe we can include loop in the if statement?

```suggestion
  if (!isEmptyArray(grantIds)) {
    tokenGrantStakingEvents = await grantContract.getPastEvents(
      "TokenGrantStaked",
      {
        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.grantContract,
        filter: { grantId: grantIds },
      }
    )

    for (const grantStakedEvent of tokenGrantStakingEvents) {
      const operator = grantStakedEvent.returnValues.operator
      const grantId = grantStakedEvent.returnValues.grantId
      if (operatorToDetails.hasOwnProperty(operator)) {
        operatorToDetails[operator].grantId = grantId
        operatorToDetails[operator].isFromGrant = true
      }
    }
  }
```